stix_shifter_modules/stix_bundle/stix_transmission/connector.py

```diff
@@ -31,19 +30,7 @@ def __init__(self, connection, configuration):
     # returns the first for some reason
     def match(self, pattern, observed_data_sdos, verbose=False):
         compiled_pattern = Pattern(pattern)
-        matcher = MatchListener(observed_data_sdos, verbose)
-        compiled_pattern.walk(matcher)
-
-        found_bindings = matcher.matched()
-
-        if found_bindings:
-            matching_sdos = []
-            for binding in found_bindings:
-                matches = [match for match in matcher.get_sdos_from_binding(binding) if match not in matching_sdos]
-                matching_sdos.extend(matches)
-        else:
-            matching_sdos = []
```

Comment:
so the match() doesn't return just the first anymore?

Reply:
looks like this optimization introduced the original problem where the match() function doesn't return all the matching observables.

Reply:
there was such problem before?